packages/databases-collections/src/components/create-namespace-modal.spec.tsx

```diff
@@ -0,0 +1,302 @@
+            },
+            createDataKey() {
+              return Promise.resolve({});
+            },
+            configuredKMSProviders() {
+              return [];
+            },
+          };
+        },
+      });
+
+    appRegistry = globalAppRegistry;
+
+    await connectionsStore.actions.connect(mockConnection);
+
+    createCollectionSpy = sandbox.spy(
+      getDataServiceForConnection(connectionId),
+      'createCollection'
+    );
+  });
+
+  afterEach(function () {
+    sandbox.resetHistory();
+    cleanup();
+  });
+
+  context('Create collection', function () {
+    beforeEach(async () => {
+      appRegistry.emit(
+        'open-create-collection',
+        {
+          database: 'foo',
+        },
+        { connectionId: '12345' }
+      );
+
+      await waitFor(() =>
+        screen.getByRole('heading', { name: 'Create Collection' })
+      );
+    });
+
+    it('renders the correct text on the submit button', () => {
+      const submitButton = screen.getByTestId('submit-button');
+      expect(submitButton.textContent).to.equal('Create Collection');
+    });
+
+    it('button is disabled when the input is empty', () => {
+      const submitButton = screen.getByTestId('submit-button');
+      const input = screen.getByTestId('collection-name');
+      expect(submitButton.getAttribute('aria-disabled')).to.equal('true');
+
+      userEvent.type(input, 'baz');
+      expect(submitButton.getAttribute('aria-disabled')).to.equal('false');
+      userEvent.clear(input);
+      expect(submitButton.getAttribute('aria-disabled')).to.equal('true');
+    });
+
+    context('when the user has submitted the form (with options)', () => {
+      beforeEach(() => {
+        const submitButton = screen.getByRole('button', {
+          name: 'Create Collection',
+        });
+        const input = screen.getByRole('textbox', { name: 'Collection Name' });
+        const additionalPreferences = screen.getByRole('button', {
+          name: /Additional preferences/,
+        });
+        userEvent.type(input, 'bar');
+        userEvent.click(additionalPreferences);
+        const clusteredCollection = screen.getByRole('checkbox', {
+          name: 'Clustered Collection',
+        });
+        userEvent.click(clusteredCollection, undefined, {
+          // leafygreen adds pointer-events: none on actually clickable elements
+          skipPointerEventsCheck: true,
+        });
+        userEvent.click(submitButton);
+      });
+
+      it('calls the dataservice create collection method', async () => {
+        await waitFor(() => {
+          expect(createCollectionSpy).to.have.been.calledOnceWith(
+            'foo.bar',
+            Sinon.match({
+              clusteredIndex: {
+                unique: true,
+              },
+            })
+          );
+        });
+      });
+    });
+
+    context(
+      'when the user submitted the collection creation form with extra whitespaces',
+      () => {
+        beforeEach(() => {
```

Comment:
Super tiny nit: you're mixing arrow functions with "plain" ones here, mocha recommends using plain instead of arrow (because `this` in mocha provides access to context, although we rarely use it) and I think that's what we're generally trying to stick with. I was actually sure we had an eslint rule for that, but seems like we didn't 😆 and I see arrows being used in other places, so totally don't feel too pressured to chage it: I'll see if there was actually a rule that we just forgot to turn on and whether or not it will have an autofix that we can apply